### DISCUSSION TITLE
Independent Combinator does not support

### DISCUSSION THREAD

**Discussion by chandana17chandana** [react-querybuilder/react-querybuilder#600] [Q&A] [answered]
"Why is the independent combinator not displaying the combinator for version 6?"

**Comment by jakeboone02** [comment]
I'm not sure exactly what the issue is without more information, but I think maybe you're not seeing the combinators because there are fewer than two rules? When `independentCombinators` is enabled (or `showCombinatorsBetweenRules`, for that matter), a combinator selector only shows up in groups that have more than one child object (rule or group).

One rule:

![image](https://github.com/react-querybuilder/react-querybuilder/assets/366438/87a6585b-2c29-409f-ac25-d1bca9bc83bb)

Two rules:

![image](https://github.com/react-querybuilder/react-querybuilder/assets/366438/be5df461-e1c6-4149-bea6-db91f74f7c18)

**Reply by chandana17chandana** [reply]
Hi jakeboone02,

Here is my code:

<QueryBuilder
	fields={subFields}
	query={query}
	onQueryChange={(newQuery) => {
		// ......	
	}}
	
	controlClassnames={{
		queryBuilder: "queryBuilder-branches",
	}}
	independentCombinators
	combinators={customCombinators}
	controlElements={{
		fieldSelector: FieldSelector,
		valueEditor: CustomValueEditor,
	}}
/>;

************ I am using version 6.5 querybuilder
I am not able to see any combinator in the query builder when i use the above code

**Reply by chandana17chandana** [reply]
<img width="845" alt="image" src="https://github.com/react-querybuilder/react-querybuilder/assets/61467865/6f3a9712-e9ea-422f-ba40-cc634db6cc6b">

**Reply by jakeboone02** [reply]
What does the `query` object look like?

**Reply by chandana17chandana** [reply]
Below is the query format :

{
    "condition": "",
    "combinator": "",
    "rules": [
        {
            "id": "d18e7d66-055f-4605-b769-a1e277d9cb2e",
            "field": "",
            "operator": "=",
            "valueSource": "value",
            "value": ""
        }
    ],
    "id": "17737593-38d9-464e-afdc-186deb794297"
}

**Reply by chandana17chandana** [reply]
Does independent combinators will support the version 6.5 ?

**Reply by jakeboone02** [reply, marked as answer]
> Below is the query format :
> 
> 
> 
> {
> 
>     "condition": "",
> 
>     "combinator": "",
> 
>     "rules": [
> 
>         {
> 
>             "id": "d18e7d66-055f-4605-b769-a1e277d9cb2e",
> 
>             "field": "",
> 
>             "operator": "=",
> 
>             "valueSource": "value",
> 
>             "value": ""
> 
>         }
> 
>     ],
> 
>     "id": "17737593-38d9-464e-afdc-186deb794297"
> 
> }

The `combinator` property is what's messing it up. Remove `combinator` from your query object and it should display correctly.

**Reply by chandana17chandana** [reply]
But when using custom combinators its throwing error...for default combinators its working fine as you said

**Reply by chandana17chandana** [reply]
The error is something like this ....
Cannot use 'in' operator to search for 'rules' in undefined
TypeError: Cannot use 'in' operator to search for 'rules' in undefined

**Reply by chandana17chandana** [reply]
const customCombinators = [
		{ label: "and", value: "and" },
		{ label: "or", value: "or" },
		{ label: "or not", value: "or not" },
		{ label: "and not", value: "and not" }, // Add your custom combinator
	];

<QueryBuilder
													fields={subFields}
													query={query}
													onQueryChange={(newQuery) => {
														// Call the addRule function to update the query.

														dispatch(setTableName("sub"));
														addRule(newQuery, setQuery);
													}}
													//combinators={customCombinators}													showCombinatorsBetweenRules
													independentCombinators
												
													controlClassnames={{
														queryBuilder: "queryBuilder-branches",
														
													}}


If i comment combinators prop its working fine but its throwing error when i use custom combinator prop as i said

**Reply by chandana17chandana** [reply]
Thanks @jakeboone02 , got resolved
